### PR TITLE
Restrict Scheduler resources

### PR DESCRIPTION
**Pull request by 2uasimojo**
Add an admission webhook restricting regular users from operating on Scheduler resources.

[OSD-6647](https://issues.redhat.com/browse/OSD-6647)

### CODE REVIEW

**Review by lisa — CHANGES_REQUESTED** [review]
Please run `make syncset` and commit/push the resulting update to `build/selectorsyncset.yaml`

Otherwise, code looks good 👍

**Review by lisa — APPROVED** [review]

**Comment by 2uasimojo** [comment]
Tested. Works.

/hold cancel

**Comment by 2uasimojo** [comment]
[Promotion MR](https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/15831)

File: pkg/webhooks/regularuser/regularuser_test.go [hunk 1]
```diff
@@ -280,6 +280,39 @@ func TestNodesSubjectPermissionsClusterVersions(t *testing.T) {
...
+			operation:       v1beta1.Update,
+			shouldBeAllowed: true,
+		},
+		{
+			testID:          "schedulers-priv-user",
+			targetResource:  "schedulers",
+			targetKind:      "Scheduler",
+			targetVersion:   "v1",
+			targetGroup:     "",
+			username:        "my-user",
```
Comment (superfunhappytime) [inline pkg/webhooks/regularuser/regularuser_test.go:300]: I see my-name vs my-user in the structures for some tests - I assume this doesn't matter?

Comment (superfunhappytime) [reply pkg/webhooks/regularuser/regularuser_test.go:300]: Ah, yes - I see it in other places; Looks fine

Comment (2uasimojo) [reply pkg/webhooks/regularuser/regularuser_test.go:300]: Yeah, the only significance is that it's not a "special" username like `kube:admin`.

**Pull request merged**